File: exercises/exercise-1-testing-java/src/main/java/de/emaarco/example/adapter/outbound/db/bike/BikeSubscriptionEntityMapper.java
package de.emaarco.example.adapter.outbound.db.bike;

import de.emaarco.example.domain.Email;
import de.emaarco.example.domain.Name;
import de.emaarco.example.domain.bike.BikeId;
import de.emaarco.example.domain.bike.BikeSubscription;
import de.emaarco.example.domain.bike.BikeSubscriptionId;
import org.springframework.stereotype.Component;

@Component
public class BikeSubscriptionEntityMapper {

    public BikeSubscription toDomain(BikeSubscriptionEntity entity) {
        return new BikeSubscription(
            new BikeSubscriptionId(entity.getId()),
            new BikeId(entity.getBikeId()),
            new Email(entity.getCustomerEmail()),
            new Name(entity.getCustomerName()),
            entity.getRequestDate(),
            entity.getStatus(),
            entity.getBikeAvailable()
        );
    }

    public BikeSubscriptionEntity toEntity(BikeSubscription domain) {
        return new BikeSubscriptionEntity(
            domain.id().value(),
            domain.bikeId().value(),
            domain.customerEmail().value(),
            domain.customerName().value(),
            domain.status(),
            domain.requestDate(),
            domain.bikeAvailable()
        );
    }

}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/de/emaarco/example/adapter/outbound/db/bike/BikeSubscriptionEntityMapper.kt
package de.emaarco.example.adapter.outbound.db.bike

import de.emaarco.example.domain.Email
import de.emaarco.example.domain.Name
import de.emaarco.example.domain.bike.BikeId
import de.emaarco.example.domain.bike.BikeSubscription
import de.emaarco.example.domain.bike.BikeSubscriptionId
import org.springframework.stereotype.Component

@Component
class BikeSubscriptionEntityMapper {

    fun toDomain(entity: BikeSubscriptionEntity): BikeSubscription {
        return BikeSubscription(
            id = BikeSubscriptionId(entity.id),
            bikeId = BikeId(entity.bikeId),
            customerEmail = Email(entity.customerEmail),
            customerName = Name(entity.customerName),
            requestDate = entity.requestDate,
            status = entity.status,
            bikeAvailable = entity.bikeAvailable
        )
    }

    fun toEntity(domain: BikeSubscription): BikeSubscriptionEntity {
        return BikeSubscriptionEntity(
            id = domain.id.value,
            bikeId = domain.bikeId.value,
            customerEmail = domain.customerEmail.value,
            customerName = domain.customerName.value,
            requestDate = domain.requestDate,
            status = domain.status,
            bikeAvailable = domain.bikeAvailable
        )
    }
}
